File: api/schemas/onboarding.py
from pydantic import BaseModel


class OnboardingStatus(BaseModel):
    has_accounts: bool
    has_incomes: bool
    has_debts: bool
    has_recurring_bills: bool
    accounts_count: int
    incomes_count: int
    debts_count: int
    recurring_bills_count: int
    completeness_score: float  # 0.0..1.0 — LEGACY (kept for existing consumers)
    # Phase 8 B2: activation is the new "ready to use it daily" gate.
    # is_activated == has_accounts AND has_balance AND has_expense.
    is_activated: bool
    has_balance: bool
    has_expense: bool


class CategoriesResponse(BaseModel):
    categories: list[str]

File: mobile/src/api/onboarding.ts
import { api } from "./client";

/**
 * Phase 8 B2: activation status. `is_activated` (1 account + a real balance + 1
 * expense) gates the first-run UX. `completeness_score` is legacy.
 */
export interface OnboardingStatus {
  has_accounts: boolean;
  has_incomes: boolean;
  has_debts: boolean;
  has_recurring_bills: boolean;
  accounts_count: number;
  incomes_count: number;
  debts_count: number;
  recurring_bills_count: number;
  completeness_score: number;
  is_activated: boolean;
  has_balance: boolean;
  has_expense: boolean;
}

export async function fetchOnboardingStatus(): Promise<OnboardingStatus> {
  const { data } = await api.get<OnboardingStatus>("/onboarding/status");
  return data;
}
